csrc/scheduler/mma_utils.cpp

```diff
@@ -705,9 +705,42 @@ std::unordered_set<IterDomain*> getMmaDomainSet(
 
 } // namespace
 
+bool isLdMatrixTranspose(const LoadStoreOp* ldst) {
+  const auto consumer = ldst->out()->isA<TensorView>()
+      ? ldst->out()->as<TensorView>()
+      : ldst->out()->as<kir::TensorIndex>()->view();
+  const auto producer = ldst->in()->isA<TensorView>()
+      ? ldst->in()->as<TensorView>()
+      : ldst->in()->as<kir::TensorIndex>()->view();
+
+  // Get all the IDs from the innermost ID of the allocation domain of
+  // the consumer to the root domain of the consumer.
+  const auto vals = DependencyCheck::getAllValsBetween(
+      {consumer->getMaybeRootDomain().begin(),
+       consumer->getMaybeRootDomain().end()},
+      {consumer->getMaybeAllocationDomain().back()});
+
+  const auto ids = ir_utils::filterByType<IterDomain>(vals);
+  const auto ids_on_path = std::vector<IterDomain*>(ids.begin(), ids.end());
+  NVF_CHECK(!ids_on_path.empty());
+  // This gives us the ID in the consumer root domain.
+  // We'll later map this ID to one in the producer.
```

Comment:
Isn't it possible for there to be multiple consumer root IDs that are dependencies of the innermost allocation ID? For example, if we have multiple K dimensions, we could merge them before we do the tiling splits. In such case, we need to take the innermost of the IDs that were merged.